src/userEvent/touchEvents.ts

```diff
@@ -0,0 +1,53 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import act from '../act';
+import { getHostParent } from '../helpers/component-tree';
+import { isHostElementPointerEventEnabled } from '../helpers/isHostElementPointerEventEnabled';
+
+const defaultPressEvent = {
+  persist: jest.fn(),
+  nativeEvent: {
+    timestamp: new Date().getTime(),
+  },
+  currentTarget: { measure: jest.fn() },
+};
+
+export type PressOptions = {
+  pressDuration: number;
+};
+
+export const press = (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 0 }
+) => {
+  if (isEnabledTouchResponder(element)) {
+    triggerPressEvent(element, options);
+    return;
+  }
+
+  const hostParentElement = getHostParent(element);
+  if (!hostParentElement) {
+    return;
+  }
+
+  press(hostParentElement, options);
+};
+
+const triggerPressEvent = (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 0 }
+) => {
+  act(() => {
+    element.props.onResponderGrant(defaultPressEvent);
+    jest.advanceTimersByTime(options.pressDuration);
```

Comment:
This seems to assume that user uses fake timers. Would we be able to support both fake timers and real timers?

Reply:
In order to make it work with real timers, we'd need to make it asynchronous and wait for some time. It's a more accessible API but I don't think we should recommend it. If it takes 500ms to trigger a longPress, that's a huge problem that I think the API should not permit. Triggering events that depend on time should require using fake timers imo and I believe we should promote that

Reply:
I agree that using fake timers is probably more useful/recommended approach, but I would be wary of not supporting users who would want to use real times with User Event.

The default touch duration for regular press could be kept at a low value (e.g. 50 ms). It seems that RTL User Event is having a `delay` option for specifying similar idea delay between keystrokes, etc and they use timeout of 0 ms by default. (https://testing-library.com/docs/user-event/options#delay)

Long press would probably used much more rarely than regular `press`.